pkg/azure/api/validation/validation.go

```diff
@@ -64,8 +64,11 @@ func ValidateProviderSecret(secret *corev1.Secret) field.ErrorList {
 		allErrs = append(allErrs, field.Required(secretDataPath.Child("clientID"), "must provide clientID"))
 	}
 
-	if utils.IsEmptyString(string(secret.Data[api.ClientSecret])) && utils.IsEmptyString(string(secret.Data[api.AzureClientSecret])) && utils.IsEmptyString(string(secret.Data[api.AzureAlternativeClientSecret])) {
-		allErrs = append(allErrs, field.Required(secretDataPath.Child("clientSecret"), "must provide clientSecret"))
+	if utils.IsEmptyString(string(secret.Data[api.ClientSecret])) &&
```

Comment:
According to the comment in the `types.go` file, `clientSecret` and `workloadIdentity` are mutually exclusive. Can you please add a validation for that? I know that `GetDefaultTokenCredentials` will give precedence to `workloadIdentity`, but it would still be nice to have validation for the same. Please Update the unit tests as well.

Reply:
Thanks, for reviewing @rishabh-11 ! The comment should be addressed with https://github.com/gardener/machine-controller-manager-provider-azure/pull/167/commits/9f2e963fa5c9a6dfd084f78b5aa4681ba626c461